app/javascript/spec/workflow-credential-mapping-form/workflow-credential-mapping-form.spec.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import toJson from 'enzyme-to-json';
+import fetchMock from 'fetch-mock';
+
+import { act } from 'react-dom/test-utils';
+import { mount } from '../helpers/mountForm';
+import WorkflowCredentialMappingForm from '../../components/workflow-credential-mapping-form/index';
+
+describe('Workflow Credential Form Component', () => {
+  const authenticationsApi = {
+    resources: [
+      { name: 'Test Credential', ems_ref: 'test_credential' },
+      { name: 'API Credential', ems_ref: 'api_credential' },
+    ],
+  };
+
+  const configurationScriptPayloadsApi = {
+    name: 'test-workflow.asl',
+    payload: `{
+      "States": {
+        "State1": { "Credentials": { "api_user": "$.api_user", "api_password": "$.api_password" } },
+        "State2": { "Credentials": { "api_user": "$.api_user", "api_password": "$.api_password" } }
+      }
+    }`,
+    credentials: {
+      credentials: {
+        api_user: { credential_ref: 'api_credential', credential_field: 'user' },
+        api_password: { credential_ref: 'test_credential', credential_field: 'password' },
+      },
+    },
```

Comment:
This double nesting is unexpected - is this right @agrare ?

Reply:
yeah I forgot to bring that up, I also have to remove it before submitting otherwise it keeps adding every post/patch https://github.com/ManageIQ/manageiq-ui-classic/pull/8905/files#diff-841ddd279784dfaa856c329e15f174b182f832fd7bd3cddbeb34149d3d1a9041R89

Reply:
That sounds like a bug, then.

Reply:
Oh ha just saw this after pulling this PR down and testing it, https://github.com/ManageIQ/manageiq-ui-classic/pull/8905#issuecomment-1729616650

Yes I think this form is incorrectly submitting credentials nested under credentials

Reply:
Yeah it ended up being on my end I'm pushing a fix in a sec